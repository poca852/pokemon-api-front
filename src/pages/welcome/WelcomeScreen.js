import React from 'react';
import { NavLink } from 'react-router-dom';
import logo from '../../assets/Pokemon-logo.png';
import pikachu from '../../assets/pikachu-circle.png';
import './welcome.css';

export const WelcomeScreen = () => {
  return (
    <div className='welcome__container'>
      <div className='welcome__logo'>
        <img src={logo} alt='logo de pokemon' />
        <button>
          <NavLink to={`/home`}>START</NavLink>
        </button>
      </div>
      <div className='welcome__pikachu'>
        <img src={pikachu} alt='pikachu' />
      </div>
    </div>
  )
}
