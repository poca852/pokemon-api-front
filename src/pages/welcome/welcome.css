.welcome__container {
   background-color: #fa6938;
   display: grid;
   grid-template-rows: repeat(2, 1fr);
   grid-template-areas: "logo pikachu";
   grid-template-rows: 1fr;
   grid-gap: 20px;
   /* justify-items: center; */
   align-items: center;
   overflow: hidden;
   height: 100vh;
   width: 100vw;
}

.welcome__logo {
   align-self: start;
   grid-area: logo;
}

.welcome__logo img {
   width: 100%;
}

.welcome__logo button {
   background-color: #305a98;
   border-radius: 10px;
   cursor: pointer;
   border: none;
   font-size: 20px;
   position: relative;
   padding: 7px 15px;
   top: -120px;
   left: 345px;
}

.welcome__logo button a {
   color: white;
   text-decoration: none;
}

.welcome__pikachu {
   /* align-self: center; */
   grid-area: pikachu;
}

.welcome__pikachu img {
   width: 100%;
}

@media only screen and (max-width: 600px) {
   .welcome__container {
      grid-template-rows: 1fr;
      grid-template-areas: "logo"
         "pikachu";
      grid-template-rows: repeat(2, 1fr);
      align-items: center;
      overflow: hidden;
   }

   .welcome__logo {
      align-self: center;
      grid-area: logo;
   }

   .welcome__logo img {
      position: relative;
      width: 100%;
   }

   .welcome__pikachu {
      align-self: start;
      grid-area: pikachu;
   }

   .welcome__pikachu img {
      position: relative;
      top: -185px;
   }

   .welcome__logo button {
      position: relative;
      top: 300px;
      left: 150px;
      font-size: 30px;
   }
}