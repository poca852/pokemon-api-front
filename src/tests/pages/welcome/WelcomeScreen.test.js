import React from 'react';
import {shallow} from 'enzyme'
import { WelcomeScreen } from '../../../pages/welcome/WelcomeScreen';

describe('Pruebas en <WelcomeScreen />', () => {
   test('Debe mostrar el componente WelcomeScreen correctamente', () => {
      const wrapper = shallow(<WelcomeScreen />)

      expect(wrapper).toMatchSnapshot();
   })
});
