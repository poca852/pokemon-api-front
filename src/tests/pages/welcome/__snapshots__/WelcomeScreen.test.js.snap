// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <WelcomeScreen /> Debe mostrar el componente WelcomeScreen correctamente 1`] = `
<div
  className="welcome__container"
>
  <div
    className="welcome__logo"
  >
    <img
      alt="logo de pokemon"
      src="Pokemon-logo.png"
    />
    <button>
      <NavLink
        to="/home"
      >
        START
      </NavLink>
    </button>
  </div>
  <div
    className="welcome__pikachu"
  >
    <img
      alt="pikachu"
      src="pikachu-circle.png"
    />
  </div>
</div>
`;
